src/components/TextInput/TextInput.scss

```diff
@@ -91,22 +104,23 @@ $block: '.#{variables.$ns}text-input';
         margin-top: 2px;
     }
 
-    /* increase css specificity to rewrite Button styles */
-    &__clear[class] {
-        --yc-button-background-color: transparent;
-        --yc-button-background-color-hover: transparent;
+    &__clear {
+        /* increase css specificity to rewrite Button styles */
+        &[class] {
+            --yc-button-background-color: transparent;
+            --yc-button-background-color-hover: transparent;
```

Comment:
How about not using Button component here? Let it be just `<button>` with styling. These vars are private for button

Reply:
I`ve created issue https://github.com/gravity-ui/uikit/issues/645